### ISSUE TITLE
cppcoreguidelines-interfaces-global-init false positive for global references initialized from other globals

### ISSUE THREAD

**Issue by LegalizeAdulthood** [llvm/llvm-project#123485]
Suppose you have the following code:

```
double g_params[MAX_PARAMS]{};
static const double &LAMBDA{g_params[0]};
```

`cppcoreguidelines-interfaces-global-init` is reported for the declaration of `LAMBDA`, but this is erroneous.

To reproduce:

1. `git clone https://github.com/LegalizeAdulthood/iterated-dynamics.git`
2. `git checkout 3f219d4dfce69d8bcab73412d0b261f378a6a239`
3. `cmake --preset default`
4. run clang-tidy on `lorenz.cpp`
5. false positive reported for lines 1295-1300

**Comment by gxyd** [comment]
Hi,

I tried creating an MRE from the specified repository, and below is the MRE I got:
```c++
extern double                g_params[];     // this is present on line 86 of `libid/include/engine/id_data.h`
static const double &LAMBDA{g_params[0]};       // this is present on line 1295 of `libid/fractals/lorenz.cpp`

```

the warning I get is:
```
(llvm-dev) [~/OpenSource/llvm-project/build][main] 13:01$ ./bin/clang-tidy ~/Documents/clang-tidy-examples/tidy_123485.cpp -checks=cppcoreguidelines-interfaces-global-init -- -std=c++20
1 warning generated.
/Users/gxyd/Documents/clang-tidy-examples/tidy_123485.cpp:6:22: warning: initializing non-local variable with non-const expression depending on uninitialized non-local variable 'g_params' [cppcoreguidelines-interfaces-global-init]
    6 | static const double &LAMBDA{g_params[0]};
      |                      ^
```

which seems like a legit warning to me.

The definition below isn't visible to the translation unit that triggers the warning (in file `libid/ui/id_main.cpp`):
```cpp
double g_params[MAX_PARAMS]{};                                         // parameters
```

perhaps that's the case?

Kindly let me know if I'm wrong in my analysis here somewhere. Thanks.
